src/operator/softmax_output.cc

```diff
@@ -30,59 +30,66 @@ Operator *SoftmaxOutputProp::CreateOperatorEx(Context ctx, std::vector<TShape> *
 DMLC_REGISTER_PARAMETER(SoftmaxOutputParam);
 
 MXNET_REGISTER_OP_PROPERTY(SoftmaxOutput, SoftmaxOutputProp)
-.describe(R"code(Softmax with logit loss.
+.describe(R"code(Computes the gradient of cross entropy loss with respect to softmax output.
 
-In the forward pass, the softmax output is returned. Assume the input data has
-shape *(n,k)*, then the output will have the same shape as the input, which is computed by
+- Softmax Function:
 
 .. math::
-   out[i,:] = softmax(data[i,:])
 
-for :math:`i=0,...,n-1`, where
+  \text{softmax}(x)_i = \frac{exp(x_i)}{\sum_j exp(x_j)}
+
+- Cross Entropy Function:
+
+.. math::
+
+  \text{CE(label, output)} = - \sum_i \text{label}_i \log(\text{output}_i)
+
+- The gradient of cross entropy loss w.r.t softmax output:
 
 .. math::
-   softmax(x) = \left[..., \frac{exp(x[j])}{exp(x[0])+...+exp(x[k-1])}, ...\right]
 
-For general *N*-D input array with shape :math:`(d_1, ..., d_n)`. Denoted by the size
-:math:`s=d_1d_2...d_n`. The way to compute softmax various:
+  \text{gradient} = \text{output} - \text{label}
 
-- ``preserve_shape`` is false (default). Reshape input into a 2-D array with
-  shape :math:`(d_1, s/d_1)` beforing computing the softmax, and then reshaped back to the
-  original shape.
+- In the forward propagation, the softmax function is computed for each instance in the input array.
 
-- ``preserve_shape`` is true. For all :math:`i_1, ..., i_{n-1}`, compute
+  For general *N*-D input arrays with shape :math:`(d_1, d_2, ..., d_n)`. The size is
+  :math:`s=d_1 \cdot d_2 \cdot \cdot \cdot d_n`. We can use the parameter ``preserve_shape``
+  and ``multi_output`` to specify the way to compute softmax:
 
-  .. math::
-    out[i_1, ..., i_{n-1}, :] = softmax(data[i_1, ..., i_{n-1},:])
+  By Default, ``preserve_shape`` is ``false``. This operator will reshape the input array
```

Comment:
preserve_shape and multi_output should both be in single backticks. Use single backticks for all variable names, function names, parameter names, etc.